### PR TITLE
cifs for cloud backups on lan

### PR DESCRIPTION
**Pull request by MattDHill**

### CODE REVIEW

File: setup-wizard/src/app/modals/cifs-modal/cifs-modal.page.html [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      ></ion-input>
+    </ion-item>
+    <p [hidden]="username.valid || username.pristine">
+      <ion-text color="danger">Username is required</ion-text>
+    </p>
+
+    <p>Password</p>
+    <ion-item>
+      <ion-input
+        id="passwword"
```
Comment (DrewAns) [inline setup-wizard/src/app/modals/cifs-modal/cifs-modal.page.html:59]: two w's.  Also should the type of this input be password so its ******

File: setup-wizard/src/app/pages/recover/recover.page.ts [hunk 2]
```diff
@@ -32,25 +34,43 @@ export class RecoverPage {
...
+  driveClickable (drive: MappedDisk) {
+    return drive['embassy-os']?.full && (this.stateService.hasProductKey || this.is02x(drive))
   }
 
   async getDrives () {
+    this.driveTargets = []
     try {
       const drives = await this.apiService.getDrives()
-      this.drives = drives.filter(d => d.partitions.length)
+      drives.filter(d => d.partitions.length).forEach(d => {
```
Comment (DrewAns) [inline setup-wizard/src/app/pages/recover/recover.page.ts:49]: maybe a good place for a map?

**Review by dr-bonez — CHANGES_REQUESTED** [review]

**Review by dr-bonez — CHANGES_REQUESTED** [review]

File: setup-wizard/src/app/services/api/api.service.ts [hunk 3]
```diff
@@ -6,8 +6,9 @@ export abstract class ApiService {
   abstract getRecoveryStatus (): Promise<RecoveryStatusRes> // setup.recovery.status
 
   // encrypted
+  abstract verifyCifs (cifs: VerifyCifs): Promise<EmbassyOSRecoveryInfo> // echo - throws error if invalid
```
[the remark this answers is not in the document]
Comment (MattDHill) [reply setup-wizard/src/app/services/api/api.service.ts:9]: This is a outdated comment on the abstract class. The actual method being hit currently is `setup.cifs.check`, as seen in live-api.service.ts. I can change to `steup.cifs.verify` if you prefer.

File: setup-wizard/src/app/services/api/api.service.ts [hunk 4]
```diff
@@ -6,8 +6,9 @@ export abstract class ApiService {
...
   abstract verifyProductKey (): Promise<void> // echo - throws error if invalid
-  abstract verify03XPassword (logicalname: string, password: string): Promise<boolean> // setup.recovery.test-password
+  abstract verify03XPassword (source: RecoverySource, password: string): Promise<boolean> // setup.recovery.test-password
```
[the remark this answers is not in the document]
Comment (MattDHill) [reply setup-wizard/src/app/services/api/api.service.ts:11]: Tech debt. Will delete.

File: setup-wizard/src/app/services/api/api.service.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (MattDHill) [reply setup-wizard/src/app/services/api/api.service.ts:9]: Is this still valid?

`abstract verifyProductKey (): Promise<void> // echo - throws error if invalid`

**Pull request closed without merging**